Projects/Core/Sources/Service/SearchHistoryDataManager/ArtistDataManager.swift

```diff
@@ -20,49 +20,50 @@ public final class ArtistDataManager {
     var newArtistInfo: ArtistInfo = artistInfo
     var songList: [String] = []
     
-    dataService.searchArtistFromGenius(artistName: newArtistInfo.name ) { result in
-      if let result = result {
-        DispatchQueue.main.async {
-          newArtistInfo = self.findArtistIdAndImage(artistInfo: newArtistInfo, hits: result.response?.hits ?? [])
-          
-          self.fetchAllSongs(artistId: newArtistInfo.gid ?? 0) { result in
-            if let result = result {
-              songList = result
-              for song in songList {
-                parsedSongList.append(Titles(title: self.extractTextBeforeParentheses(from: song), subTitle: self.extractTextInsideFirstParentheses(from: song) ?? ""))
-              }
-              newArtistInfo.songList = parsedSongList
-              completion(newArtistInfo)
-            } else {
-              completion(nil)
-            }
-          }
-          completion(newArtistInfo)
-        }
-      } else {
-        completion(nil)
-      }
-    }
+	  dataService.callRequest(type: GeniusArtistsModel.self, api: .searchGenius(name: newArtistInfo.name)) { result in
+		  if let result = result {
+			 DispatchQueue.main.async {
+				newArtistInfo = self.findArtistIdAndImage(artistInfo: newArtistInfo, hits: result.response?.hits ?? [])
+
+				self.fetchAllSongs(artistId: newArtistInfo.gid ?? 0) { result in
+				  if let result = result {
+					 songList = result
+					 for song in songList {
+						parsedSongList.append(Titles(title: self.extractTextBeforeParentheses(from: song), subTitle: self.extractTextInsideFirstParentheses(from: song) ?? ""))
+					 }
+					 newArtistInfo.songList = parsedSongList
+					 completion(newArtistInfo)
+				  } else {
+					 completion(nil)
+				  }
+				}
+				completion(newArtistInfo)
+			 }
+		  } else {
+			 completion(nil)
+		  }
```

Comment:
뎁스를 더 줄여보고, async completion handler 에서 메모리 캡쳐를 통해 안전하게 접근 하는 것이 좋아보입니다. weak self 없이 바로 캡쳐하면 reference cycle 이 발생합니다.
또한 handler 안에 handler 를 반복하기 보다는 적절하게 값을 밖으로 빼내고 변수를 미리 선언하기 보다는 정말 필요한 시점에서 변수선언을 하는 것이 어떨까요?

로직 이해가 없는 상태로 작성한 suggestion 이라 맘에 드는 코드는 아니지만 조심스레 제안드려봅니다.
```suggestion
		  guard let result = result else { 
		      completion(nil) 
		      return
		  }
		  DispatchQueue.main.async { [weak self, newArtistInfo] in
		      guard let self else { 
		          completion(nil)
		          return
		      }
		      newArtistInfo = self.findArtistIdAndImage(artistInfo: newArtistInfo, hits: result.response?.hits ?? [])

			  self.fetchAllSongs(artistId: newArtistInfo.gid ?? 0) { result in
				  guard let result = result else {
				      completion(nil)
				      return
				  }
				  let parsedSongList  = result.compactMap { 
				      Titles(
				          title: self.extractTextBeforeParentheses(from: $0),
				          subTitle: self. self.extractTextInsideFirstParentheses(from: $0) ?? ""
				      )
				  }
				  newArtistInfo.songList = parsedSongList
				  completion(newArtistInfo)
		      }
		  }
```